Gifski/Utilities.swift

```diff
@@ -595,6 +595,102 @@ extension AVAsset {
 }
 
 
+extension AVAssetTrack {
+	enum VideoTrimmingError: Error {
+		case assetIsNil
+		case assetReaderFailure
+		case assetIsEmpty
+		case assetIsMissingTrack
+		case compositionCouldNotBeCreated
+	}
+
+	/**
+	Removes blank frames from the begining of the track.
+
+	This can be useful to trim blank frames from files produced by tools like the iOS simulator screen recorder.
+	*/
+	func trimmingBlankFrames() throws -> AVAssetTrack {
+		// Create new composition
+		let composition = AVMutableComposition()
+		guard
+			let wrappedTrack = composition.addMutableTrack(withMediaType: mediaType, preferredTrackID: .zero)
+		else {
+			throw VideoTrimmingError.compositionCouldNotBeCreated
+		}
+		try wrappedTrack.insertTimeRange(timeRange, of: self, at: .zero)
+
+		let reader = try AVAssetReader(asset: composition)
+
+		// Create reader for wrapped track.
+		let readerOutput = AVAssetReaderTrackOutput(track: wrappedTrack, outputSettings: nil)
+		reader.add(readerOutput)
+		reader.startReading()
+
+		defer {
+			reader.cancelReading()
+		}
+
+		// Iterate through samples until we reach one with a non-zero size.
+		while let sampleBuffer = readerOutput.copyNextSampleBuffer() {
+			guard [.completed, .reading].contains(reader.status) else {
+				throw reader.error ?? VideoTrimmingError.assetReaderFailure
+			}
+
+			// On first non-empty frame.
+			guard sampleBuffer.totalSampleSize == 0 else {
+				let currentTimestamp = sampleBuffer.outputPresentationTimeStamp
+				wrappedTrack.removeTimeRange(.init(start: .zero, end: currentTimestamp))
+				return wrappedTrack
+			}
+		}
+
+		throw VideoTrimmingError.assetIsEmpty
```

Comment:
This error is misnamed. It should be `videoTrackIsEmpty`.

Reply:
Fixed in 9a81320.